### DISCUSSION TITLE
建议添加获取某个时间戳之后的所有消息的 API

### DISCUSSION THREAD

**Discussion by Steve-xmh** [howmanybots/onebot#145] [想法 / 建议]
如题，如果遇到某些不可抗因素导致需要重启，重启期间的消息需要处理，可能会需要用到这个 API 获取历史消息

**Comment by ExerciseBook** [comment]
@richardchien 这个问题要不要加入到 V12 的讨论范围内？

**Comment by stdrc** [comment]
@ExerciseBook 这感觉没必要吧，说实话感觉存消息应该自己存，毕竟这跟业务逻辑很相关，OneBot 毕竟也不是一个真正的 IM 服务

**Comment by stdrc** [comment]
想了下这个这个需求也许应该有 OneBot 实现来决定，比如在没地方推送消息的时候，先存在缓冲区，之后等再次连接后再推送，似乎没必要专门搞个 API

**Comment by Steve-xmh** [comment]
> 想了下这个这个需求也许应该有 OneBot 实现来决定，比如在没地方推送消息的时候，先存在缓冲区，之后等再次连接后再推送，似乎没必要专门搞个 API

如果遇到多个连接的话，实现就没办法很好确定各客户端最后接收的消息了，例如聊天和加群事件（这个比较重要）

再出这样一个api的确不是bot的样子（雾）不过如果有方法在不变更当前api来继续上一次连接到现在重新连接期间没处理的消息的话就最好了，如果按着这个方向考虑的话，就得需要一个方式来确认识别每一个连接客户端并保存各个客户端最后连接的状态

**Comment by stdrc** [comment]
还是觉得这个不属于接口标准所定义的范围，一个优秀的实现可以自行判断哪个客户端最后拿到了哪个消息，一般的实现如果不考虑这个，在大部分场景也基本能用，先 close 了，之后如果有更好的 idea 再打开
